crates/biome_js_analyze/src/analyzers/nursery/use_shorthand_assign.rs

```diff
@@ -0,0 +1,268 @@
+    ///
+    /// Source: https://eslint.org/docs/latest/rules/operator-assignment/
+    ///
+    /// ## Examples
+    ///
+    /// ### Invalid
+    ///
+    /// ```js,expect_diagnostic
+    /// var a = 1;
+    /// a = a + 1;
+    /// ```
+    ///
+    /// ```js,expect_diagnostic
+    /// var a = 1;
+    /// a = a - 1;
+    /// ```
+    ///
+    ///  ```js,expect_diagnostic
+    /// var a = 1;
+    /// a = a * 1;
+    /// ```
+    ///
+    ///  ```js,expect_diagnostic
+    /// var a = 1;
+    /// a = a / 1;
+    /// ```
+    ///
+    /// ```js,expect_diagnostic
+    /// var a = 1;
+    /// a = a % 1;
+    /// ```
+    ///
+    /// ```js,expect_diagnostic
+    /// var a = 1;
+    /// a = a << 1;
+    /// ```
+    ///
+    /// ```js,expect_diagnostic
+    /// var a = 1;
+    /// a = a >> 1;
+    /// ```
+    ///
+    /// ```js,expect_diagnostic
+    /// var a = 1;
+    /// a = a >>> 1;
+    /// ```
+    ///
+    /// ```js,expect_diagnostic
+    /// var a = 1;
+    /// a = a & 1;
+    /// ```
+    ///
+    /// ```js,expect_diagnostic
+    /// var a = 1;
+    /// a = a ^ 1;
+    /// ```
+    ///
+    /// ```js,expect_diagnostic
+    /// var a = 1;
+    /// a = a | 1;
+    /// ```
+    ///
+    /// ```js,expect_diagnostic
+    /// var a = 1;
+    /// a = a ** 1;
+    /// ```
+    ///
+    /// ## Valid
+    ///
+    /// ```js
+    /// var a = 1;
+    /// a += 1;
+    /// ```
+    ///
+    /// ```js
+    /// var a = 1;
+    /// a -= 1;
+    /// ```
+    ///
+    ///  ```js
+    /// var a = 1;
+    /// a *= 1;
+    /// ```
+    ///
+    ///  ```js
+    /// var a = 1;
+    /// a /= 1;
+    /// ```
+    ///
+    /// ```js
+    /// var a = 1;
+    /// a %= 1;
+    /// ```
+    ///
+    /// ```js
+    /// var a = 1;
+    /// a <<= 1;
+    /// ```
+    ///
+    /// ```js
+    /// var a = 1;
+    /// a >>= 1;
+    /// ```
+    ///
+    /// ```js
+    /// var a = 1;
+    /// a >>>= 1;
+    /// ```
+    ///
+    /// ```js
+    /// var a = 1;
+    /// a &= 1;
+    /// ```
+    ///
+    /// ```js
+    /// var a = 1;
+    /// a ^= 1;
+    /// ```
+    ///
+    /// ```js
+    /// var a = 1;
+    /// a |= 1;
+    /// ```
+    ///
+    /// ```js
+    /// var a = 1;
+    /// a **= 1;
+    /// ```
+    ///
+    pub(crate) UseShorthandAssign {
+        version: "next",
+        name: "useShorthandAssign",
+        recommended: false,
+    }
+}
+
+pub struct RuleState {
+    shorthand_operator: JsSyntaxKind,
+    right: AnyJsExpression,
+}
+
+impl Rule for UseShorthandAssign {
+    type Query = Ast<JsAssignmentExpression>;
+    type State = RuleState;
+    type Signals = Option<Self::State>;
+    type Options = ();
+
+    fn run(ctx: &RuleContext<Self>) -> Self::Signals {
+        let node = ctx.query();
+        let operator = node.operator().ok()?;
+        let right = node.right().ok();
+        let left = node.left().ok();
+        let left_var_name = left?.text();
+
+        if !matches!(operator, JsAssignmentOperator::Assign) {
```

Comment:
`operator` is not used after this check, we could avoid declaring a variable used only once.
```suggestion
        if !matches!(node.operator(), Ok(JsAssignmentOperator::Assign)) {
```